src/DropdownButton.jsx

```diff
@@ -111,6 +111,11 @@ var DropdownButton = React.createClass({
     e.preventDefault();
 
     this.setDropdownState(!this.state.open);
+    if(this.props.onClick && this.props.noCaret ){
```

Comment:
Why should `noCaret` be set to `true` in order to execute the `onClick` handler?